tests/mocha/insertion_marker_manager_test.js

```diff
@@ -0,0 +1,329 @@
+            const stub = stubComponentManager(this.workspace);
+            stub.withArgs('fakeDragTarget',
+                Blockly.ComponentManager.Capability.DELETE_AREA).returns(false);
+            const fakeDragTarget = {
+                wouldDelete: sinon.fake.returns(false),
+                id: 'fakeDragTarget',
+            };
+            this.manager.update(dxy, fakeDragTarget);
+            chai.assert.isFalse(this.manager.wouldDeleteBlock());
+            chai.assert.isFalse(fakeDragTarget.wouldDelete.called);
+        });
+
+        test('Not over drag target', function() {
+            const dxy = new Blockly.utils.Coordinate(0, 0);
+            this.manager.update(dxy, null);
+            chai.assert.isFalse(this.manager.wouldDeleteBlock());
+        });
+    });
+
+    suite('Would connect stack blocks', function() {
+        setup(function() {
+            const state = {
+                'blocks': {
+                  'blocks': [
+                    {
+                      'type': 'stack_block',
+                      'id': 'first',
+                      'x': 0,
+                      'y': 0,
+                    },
+                    {
+                      'type': 'stack_block',
+                      'id': 'other',
+                      'x': 200,
+                      'y': 200,
+                    },
+                  ],
+                },
+              };
+            Blockly.serialization.workspaces.load(state, this.workspace);
+            this.block = this.workspace.getBlockById('first');
+            this.block.setDragging(true);
+            this.manager = new Blockly.InsertionMarkerManager(this.block);
+        });
+
+        test('No other blocks nearby', function() {
+            this.manager.update(new Blockly.utils.Coordinate(0, 0), null);
+            chai.assert.isFalse(this.manager.wouldConnectBlock());
+        });
+
+        test('Near other block: above', function() {
+            this.manager.update(new Blockly.utils.Coordinate(200, 190), null);
+            chai.assert.isTrue(this.manager.wouldConnectBlock());
+            const markers = this.manager.getInsertionMarkers();
+            chai.assert.equal(markers.length, 1);
+            const marker = markers[0];
+            chai.assert.isTrue(marker.nextConnection.isConnected());
+        });
+
+        test('Near other block: below', function() {
+            this.manager.update(new Blockly.utils.Coordinate(200, 210), null);
+            chai.assert.isTrue(this.manager.wouldConnectBlock());
+            const markers = this.manager.getInsertionMarkers();
+            chai.assert.equal(markers.length, 1);
+            const marker = markers[0];
+            chai.assert.isTrue(marker.previousConnection.isConnected());
+        });
+
+        test('Near other block: left', function() {
+            this.manager.update(new Blockly.utils.Coordinate(190, 200), null);
+            chai.assert.isTrue(this.manager.wouldConnectBlock());
```

Comment:
I'm confused about why this should be true.

Reply:
You have to get pretty close, but eventually the blocks will be close enough for a connection to be an option.
![5428040e-e0a0-4c37-b469-3538cd8f2ed3](https://user-images.githubusercontent.com/13686399/199625299-3be82e0e-fa60-41d0-84d9-a82cbe047ca2.gif)


It would not be true if the blocks were taller. Then the distance between the top connection on the moving block and bottom connection on the stationary block would be too great, even if the blocks were directly on top of each other, for them to be compatible. I deliberately did not test exactly where the edge between valid and invalid is, since that would mostly be a test of the internal math.

Reply:
So why are some of the assesrtions checking `wouldConnectBlock` but others are testing if the block is actually connected? Is that due to the "not testing the exact edge" thing?

Reply:
When approaching from the top or bottom, I'm confident that there is a correct answer: when approaching from the top, it should try to attach above. When approaching from the bottom, it should try to attach below.

When approaching from left or right, there's not obviously a correct answer, so I'm not asserting anything about it.